api/src/test/java/org/openmrs/module/appointments/service/AppointmentsServiceTest.java

```diff
@@ -102,135 +125,149 @@ public void shouldSaveAppointmentsOnlyIfUserHasManagePrivilege() throws Exceptio
         assertEquals(null, appointmentsService.getAppointmentByUuid("uuid"));
     }
 
     @Test(expected = APIAuthenticationException.class)
-    public void shouldNotGetAppointmentByUuidIfUserHasNoPrivilege() throws Exception {
+    public void shouldNotGetAppointmentByUuidIfUserHasNoPrivilege() {
         Context.authenticate(noPrivilegeUser, noPrivilegeUserPassword);
         assertEquals(null, appointmentsService.getAppointmentByUuid("uuid"));
     }
 
     @Test
-    public void shouldBeAbleToChangeStatusIfUserHasManagePrivilege() throws Exception {
+    public void shouldBeAbleToChangeStatusIfUserHasManagePrivilege() {
         Context.authenticate(manageUser, manageUserPassword);
         appointmentsService.changeStatus(new Appointment(), "Completed", null);
     }
 
+    @Test
+    public void shouldBeAbleToChangeStatusIfUserHasManageOwnPrivilege() {
+        Context.authenticate(manageOwnUser, manageOwnUserPassword);
+        appointmentsService.changeStatus(new Appointment(), "Completed", null);
+    }
+
     @Test(expected = APIAuthenticationException.class)
-    public void shouldNotBeAbleToChangeStatusIfUserHasReadOnlyPrivilege() throws Exception {
+    public void shouldNotBeAbleToChangeStatusIfUserHasReadOnlyPrivilege() {
         Context.authenticate(readOnlyUser, readOnlyUserPassword);
         appointmentsService.changeStatus(new Appointment(), "Completed", null);
     }
 
     @Test
-    public void shouldGetAllAppointmentsInDateRangeIfUserHasReadOnlyPrivilege() throws Exception {
+    public void shouldGetAllAppointmentsInDateRangeIfUserHasReadOnlyPrivilege() {
         Context.authenticate(readOnlyUser, readOnlyUserPassword);
         assertNotNull(appointmentsService.getAllAppointmentsInDateRange(null, null));
     }
 
     @Test(expected = APIAuthenticationException.class)
-    public void shouldNotGetAllAppointmentsInDateRangeIfUserHasNoPrivilege() throws Exception {
+    public void shouldNotGetAllAppointmentsInDateRangeIfUserHasNoPrivilege() {
         Context.authenticate(noPrivilegeUser, readOnlyUserPassword);
         assertNotNull(appointmentsService.getAllAppointmentsInDateRange(null, null));
     }
 
     @Test(expected = APIException.class)
-    public void shouldBeAbleToUndoStatusChangeIfUserHasManagePrivilege() throws Exception {
+    public void shouldBeAbleToUndoStatusChangeIfUserHasManagePrivilege() {
         Context.authenticate(manageUser, manageUserPassword);
         Appointment appointment = new Appointment();
         appointment.setId(1);
         appointmentsService.undoStatusChange(appointment);
     }
 
+    @Test(expected = APIException.class)
+    public void shouldBeAbleToUndoStatusChangeIfUserHasManageOwnPrivilege() {
+        Context.authenticate(manageOwnUser, manageOwnUserPassword);
+        Appointment appointment = new Appointment();
+        appointment.setId(1);
+        appointmentsService.undoStatusChange(appointment);
+    }
```

Comment:
The API exception is due to the reason that there are no status change actions to undo. If the user does not have the privilege to perform undo operation the exception thrown would be authentication exception. The mock setup for appointment is not done at this class because this is just to check whether the user with appropriate privileges is able to hit the method.

Reply:
We're checking if the user has appropriate privileges can hit the method. But if the method is hit without setting the appointment context, it will throw an exception.